master/buildbot/status/worker.py

```diff
@@ -41,6 +42,30 @@ def __init__(self, name):
         self.graceful_callbacks = []
         self.pause_callbacks = []
         self.connect_times = []
+        self.info = Properties()
+
+    def setInfo(self, info):
+        self.admin = bytes2unicode(info.get("admin"))
+        self.host = bytes2unicode(info.get("host"))
+        self.access_uri = info.get("access_uri", None)
+        if self.access_uri:
+            self.access_uri = bytes2unicode(self.access_uri)
+        self.version = bytes2unicode(info.get("version", b"(unknown)"))
+
+        # store everything as Properties
+        for k, v in info.items():
```

Comment:
@tardyp do i need to decode keyvalues here?

Reply:
I'd prefer we work with unicode everywhere. So we should decode it in the worker.